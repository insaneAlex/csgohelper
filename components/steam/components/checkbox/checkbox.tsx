import React, {FC} from "react";

import styles from "./checkbox.module.scss";

type Props = {label: string; checked: boolean; onChange: () => void};

export const Checkbox: FC<Props> = ({label, checked, onChange}) => {
  return (
    <label className={styles.filters} htmlFor="">
      <input type="checkbox" checked={checked} onChange={onChange} />
      <span>{label}</span>
    </label>
  );
};
