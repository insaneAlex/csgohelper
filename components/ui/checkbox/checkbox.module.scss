@import "styles/variables.scss";

.label {
  border: 1px solid transparent;
  border-radius: 5px;
  background: $lighten-bg;
  cursor: pointer;
  display: block;
  margin: 5px;
  padding: 5px;
}

.checked {
  border: 1px solid $main-font-color;
}
