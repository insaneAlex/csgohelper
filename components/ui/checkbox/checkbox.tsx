import {FC} from "react";
import classNames from "classnames";

import styles from "./checkbox.module.scss";

type Props = {
  checked?: boolean;
  name: string;
  label: string;
  readOnly?: boolean;
  defaultChecked?: boolean;
  onChange?: React.ChangeEventHandler<HTMLInputElement>;
};

export const Checkbox: FC<Props> = ({
  checked,
  name,
  readOnly,
  label,
  defaultChecked,
  onChange,
}) => (
  <label
    className={classNames(styles.label, {
      [styles.checked]: checked || defaultChecked,
    })}
    htmlFor={name}
  >
    <input
      id={name}
      type="checkbox"
      readOnly={readOnly}
      defaultChecked={defaultChecked}
      checked={checked}
      onChange={onChange}
    />
    <span>{label}</span>
  </label>
);
