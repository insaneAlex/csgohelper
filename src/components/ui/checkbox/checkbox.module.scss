@import 'styles/index';

.label {
  background: transparent;
  border: 1px solid transparent;
  border-radius: 5px;
  cursor: pointer;
  display: flex;
  margin: 5px;
  padding: 3px 0 3px 3px;

  @include body-short;
}

.input {
  margin-right: 5px;
  cursor: pointer;
}

.checked {
  border: 1px solid $main-font-color;
}
