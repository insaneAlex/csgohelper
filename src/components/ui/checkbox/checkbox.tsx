import {FC} from 'react';
import classNames from 'classnames';

import styles from './checkbox.module.scss';

type Props = {
  name: string;
  label: string;
  checked?: boolean;
  readOnly?: boolean;
  onClick?: () => void;
  isWithoutBorder?: boolean;
  defaultChecked?: boolean;
  onChange?: React.ChangeEventHandler<HTMLInputElement>;
};

export const Checkbox: FC<Props> = ({
  name,
  label,
  checked,
  onClick,
  onChange,
  readOnly,
  defaultChecked,
  isWithoutBorder = false
}) => (
  <label
    className={classNames(styles.label, {
      [styles.checked]: (checked || defaultChecked) && !isWithoutBorder
    })}
    htmlFor={name}
  >
    <input
      id={name}
      type="checkbox"
      onClick={onClick}
      checked={checked}
      readOnly={readOnly}
      onChange={onChange}
      className={styles.input}
      defaultChecked={defaultChecked}
    />
    <span>{label}</span>
  </label>
);
