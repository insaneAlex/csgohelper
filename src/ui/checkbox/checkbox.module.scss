@import 'styles/variables.scss';

.label {
  background: $lighten-bg;
  border: 1px solid transparent;
  border-radius: 5px;
  cursor: pointer;
  display: block;
  margin: 5px;
  padding: 5px;
  display: flex;
}

.input {
  margin-right: 5px;
}

.checked {
  border: 1px solid $main-font-color;
}
