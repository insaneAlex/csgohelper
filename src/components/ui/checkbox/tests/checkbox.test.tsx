import {render} from '@testing-library/react';
import {Checkbox} from '../checkbox';

describe('Checkbox', () => {
  it('should render correctly', () => {
    const {container} = render(<Checkbox name="test" label="test1" />);
    expect(container).toMatchSnapshot();
  });

  describe('if is checked', () => {
    it('should have checked class', () => {
      const {container} = render(<Checkbox defaultChecked name="test" label="test1" />);
      const checkedLabel = container.querySelector('.label.checked');

      expect(checkedLabel).toBeInTheDocument();
    });
  });
});
