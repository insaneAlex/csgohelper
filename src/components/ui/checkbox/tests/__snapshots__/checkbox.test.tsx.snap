// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox should render correctly 1`] = `
<div>
  <label
    class="label"
    for="test"
  >
    <input
      class="input"
      id="test"
      type="checkbox"
    />
    <span>
      test1
    </span>
  </label>
</div>
`;
